bin/clover/src/spec/sockets.ts

```diff
@@ -1,18 +1,21 @@
 import { ulid } from "https://deno.land/x/ulid@v0.3.0/mod.ts";
 import { AttrFuncInputSpec } from "../bindings/AttrFuncInputSpec.ts";
 import { SocketSpec } from "../bindings/SocketSpec.ts";
+import { SocketSpecData } from "../bindings/SocketSpecData.ts";
 import { SocketSpecArity } from "../bindings/SocketSpecArity.ts";
 import { SocketSpecKind } from "../bindings/SocketSpecKind.ts";
 import { ExpandedPropSpec } from "./props.ts";
 import { getSiFuncId } from "./siFuncs.ts";
 import _ from "npm:lodash";
 import { ExpandedSchemaVariantSpec } from "./pkgs.ts";
+import { Extend } from "../extend.ts";
 
 export const SI_SEPARATOR = "\u{b}";
 
-export type ExpandedSocketSpec = SocketSpec & {
-  data: NonNullable<SocketSpec["data"]>;
-};
+export type ExpandedSocketSpec = Extend<SocketSpec, {
+  data: NonNullable<SocketSpecData>;
```

Comment:
This makes it so we bring in SocketSpecData to autocomplete even though the binding files don't import each other correctly.